### PR TITLE
Added Indexes helper

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1684

[Review on Reviewable](https://reviewable.io/reviews/rozza/mongo-java-driver/98)

### CODE REVIEW

File: driver-core/src/main/com/mongodb/client/model/Indexes.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    public static Bson compoundIndex(final List<Bson> indexes) {
+        notNull("indexes", indexes);
+        return new Bson() {
+            @Override
+            public <TDocument> BsonDocument toBsonDocument(final Class<TDocument> documentClass, final CodecRegistry codecRegistry) {
+                BsonDocument compoundIndex = new BsonDocument();
+                for (Bson index : indexes) {
+                    BsonDocument indexDocument = index.toBsonDocument(documentClass, codecRegistry);
+                    for (String key : indexDocument.keySet()) {
+                        compoundIndex.remove(key);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/model/Indexes.java]: I see that Sorts class does this too, but now I don't see why the call to remove is necessary, as append will replace any existing mapping.

**Comment by rozza** [comment]
Updated

**Comment by jyemin** [comment]
Looks good.  Only thing missing is a page for this in the reference docs under the Builders section, but feel free to push this and do a separate PR for that.

**Comment by rozza** [comment]
Its only a quick page - followed the sorts page and have added :)

**Comment by rozza** [comment]
Updated

**Comment by jyemin** [comment]
lgtm

**Pull request closed without merging**